send-transaction-service/src/send_transaction_service.rs

```diff
@@ -556,60 +548,25 @@ impl SendTransactionService {
             .unwrap()
     }
 
-    /// Process transactions in batch.
-    fn send_transactions_in_batch<T: TpuInfo>(
-        tpu_address: &SocketAddr,
-        transactions: &HashMap<Signature, TransactionInfo>,
-        leader_info: Option<&T>,
-        connection_cache: &Arc<ConnectionCache>,
-        config: &Config,
-        stats: &SendTransactionServiceStats,
-    ) {
-        // Processing the transactions in batch
-        let mut addresses = config
-            .tpu_peers
-            .as_ref()
-            .map(|addrs| addrs.iter().map(|a| (a, 0)).collect::<Vec<_>>())
-            .unwrap_or_default();
-        let leader_addresses = Self::get_tpu_addresses_with_slots(
-            tpu_address,
-            leader_info,
-            config,
-            connection_cache.protocol(),
-        );
-        addresses.extend(leader_addresses);
-
-        let wire_transactions = transactions
-            .iter()
-            .map(|(_, transaction_info)| {
-                debug!(
-                    "Sending transacation {} to (address, slot): {:?}",
-                    transaction_info.signature, addresses,
-                );
-                transaction_info.wire_transaction.as_ref()
-            })
-            .collect::<Vec<&[u8]>>();
-
-        for (address, _) in &addresses {
-            Self::send_transactions(address, &wire_transactions, connection_cache, stats);
-        }
-    }
-
     /// Retry transactions sent before.
     fn process_transactions<T: TpuInfo + std::marker::Send + 'static>(
         working_bank: &Bank,
         root_bank: &Bank,
-        tpu_address: &SocketAddr,
         transactions: &mut HashMap<Signature, TransactionInfo>,
-        leader_info_provider: &Arc<Mutex<CurrentLeaderInfo<T>>>,
-        connection_cache: &Arc<ConnectionCache>,
-        config: &Config,
+        client: &ConnectionCacheClient<T>,
+        Config {
```

Comment:
```suggestion
        &Config {
```

not sure if this works in the args list, but if so you can avoid all of the dereferences in the body. if not you can move the destructuring to top of the method with `let &Config { ... } = config`

Reply:
Magic! I didn't know about this feature, basically the following syntax:
```rust
fn function(
        &Config {
            retry_rate_ms,
            service_max_retries,
            default_max_retries,
            batch_size,
            ..
        }: &Config,
    )  {
        let bla: u64 = retry_rate_ms;
    }
```

Reply:
rust pattern syntax is pretty nice